app/screens/global_threads/threads_list/thread/thread.tsx

```diff
@@ -0,0 +1,239 @@
+            marginTop: 5,
+        },
+        mentionBadge: {
+            width: 18,
+            height: 18,
+            borderRadius: 9,
+            backgroundColor: theme.buttonBg,
+            alignSelf: 'center',
+        },
+        mentionBadgeText: {
+            ...typography('Body', 25, 'SemiBold'),
+            alignSelf: 'center',
+            color: theme.buttonColor,
+        },
+    };
+});
+
+const Thread = ({author, channel, post, teammateNameDisplay, testID, theme, thread}: Props) => {
+    const intl = useIntl();
+    const isTablet = useIsTablet();
+    const styles = getStyleSheet(theme);
+    const serverUrl = useServerUrl();
+
+    const showThread = useCallback(preventDoubleTap(() => {
+        fetchAndSwitchToThread(serverUrl, thread.id);
+    }), [serverUrl, thread.id]);
+
+    const showThreadOptions = useCallback(() => {
+        const passProps = {thread};
+        const title = isTablet ? intl.formatMessage({id: 'thread.options.title', defaultMessage: 'THREAD ACTIONS'}) : '';
+
+        if (isTablet) {
+            showModal(Screens.THREAD_OPTIONS, title, passProps, bottomSheetModalOptions(theme, 'close-thread-options'));
+        } else {
+            showModalOverCurrentContext(Screens.THREAD_OPTIONS, passProps);
+        }
+    }, [isTablet, theme, thread]);
+
+    const threadStarterName = displayUsername(author, intl.locale, teammateNameDisplay);
+    const testIDPrefix = `${testID}.${thread.id}`;
+
+    const needBadge = thread.unreadMentions || thread.unreadReplies;
+    let badgeComponent;
+    if (needBadge) {
+        if (thread.unreadMentions && thread.unreadMentions > 0) {
+            badgeComponent = (
+                <View
+                    style={styles.mentionBadge}
+                    testID={`${testIDPrefix}.unread_mentions`}
+                >
+                    <Text style={styles.mentionBadgeText}>{thread.unreadMentions > 99 ? '99+' : thread.unreadMentions}</Text>
+                </View>
+            );
+        } else if (thread.unreadReplies && thread.unreadReplies > 0) {
+            badgeComponent = (
+                <View
+                    style={styles.unreadDot}
+                    testID={`${testIDPrefix}.unread_dot`}
+                />
+            );
+        }
+    }
+
+    let name;
+    let postBody;
+    if (post.deleteAt > 0) {
+        name = (
+            <FormattedText
+                id='threads.deleted'
+                defaultMessage='Original Message Deleted'
+                style={[styles.threadStarter, styles.threadDeleted]}
+                numberOfLines={1}
+            />
+        );
+    } else {
+        name = (
+            <Text
+                style={styles.threadStarter}
+                numberOfLines={1}
+            >{threadStarterName}</Text>
+        );
+        postBody = (
+            <Text
+                style={styles.message}
+                numberOfLines={2}
+            >
+                {post?.message}
```

Comment:
If the post message is an empty string, would we still want to display the postBody?

Reply:
Not sure, but it's according to v1, it will render an empty view with just line height.

Reply:
Then, I think we can display it only if we have a message...else we're occupying real estate to display nothing.